<launch>

  <!-- local frame name -->
  <arg name="global_frame"   value="map"/>
  <arg name="odom"           value="odom"/>
  <arg name="base_footprint" value="base_footprint"/>
  <arg name="base_scan"      value="base_scan"/>

  <!-- ninebot myahrs -->
  <!--<node pkg="myahrs_driver" type="myahrs_driver" name="myahrs_driver">
    <param name="port" value="/dev/ttyACM0"/>
  </node>-->

  <!-- nucleo communication -->
  <node pkg="tms_rc_ninebot" type="ninebot_serial.py" name="ninebot_serial" output="screen">
    <param name="port" value="/dev/ttyACM1"/>
  </node>

  <!-- ninebot odometry -->
  <node pkg="tms_rc_ninebot" type="ninebot_odometry.py" name="ninebot_odometry" output="screen">
    <param name="wheel_radius"  value="0.129"/>
    <param name="odom_frame_id" value="$(arg odom)"/>
    <param name="base_frame_id" value="$(arg base_footprint)"/>
  </node>
  
  <!-- ninebot static tf -->
  <node pkg="tf" type="static_transform_publisher" name="map_to_odom"
    args="0 0 0 0 0 0 $(arg global_frame) $(arg odom) 100">
  </node>
  <node pkg="tf" type="static_transform_publisher" name="link_to_scan"
    args="0.1 0 0.36 0 0 0 $(arg base_footprint) $(arg base_scan) 100">
  </node>

  <!-- ninebot urg -->
  <arg name="urg_range" value="1.5708"/>
  <node pkg="urg_node" type="urg_node" name="urg_node" output="screen">
    <param name="frame_id" value="$(arg base_scan)"/>
    <param name="ip_address" value="192.168.0.10"/>
    <param name="angle_min" value="-$(arg urg_range)"/>
    <param name="angle_max" value="$(arg urg_range)"/>
  </node>

</launch>
